cabal-install/src/Distribution/Client/ProjectPlanOutput.hs

```diff
@@ -150,7 +150,9 @@ encodePlanAsJson distDirLayout elaboratedInstallPlan elaboratedSharedConfig =
         | Just hash <- [elabPkgSourceHash elab] ] ++
         (case elabBuildStyle elab of
             BuildInplaceOnly ->
-                ["dist-dir"   J..= J.String dist_dir]
+                [ "dist-dir"   J..= J.String dist_dir
+                , "build-info" J..= J.String (buildInfoPref dist_dir)
```

Comment:
IIRC we decided (off-list) this should be nullable to represent build-type:Custom with lib:Cabal being too old.

Reply:
Done